lib/std/os.zig

```diff
@@ -1887,7 +1887,13 @@ pub fn unlinkatZ(dirfd: fd_t, file_path_c: [*:0]const u8, flags: u32) UnlinkatEr
 /// Same as `unlinkat` but `sub_path_w` is UTF16LE, NT prefixed. Windows only.
 pub fn unlinkatW(dirfd: fd_t, sub_path_w: []const u16, flags: u32) UnlinkatError!void {
     const remove_dir = (flags & AT_REMOVEDIR) != 0;
-    return windows.DeleteFile(sub_path_w, .{ .dir = dirfd, .remove_dir = remove_dir });
+    try windows.DeleteFile(sub_path_w, .{ .dir = dirfd, .remove_dir = remove_dir });
+    if (remove_dir){
+        faccessatW(dirfd, @ptrCast([*:0]const u16, sub_path_w.ptr), 0, 0) catch 
+            return;
+        return error.DirNotEmpty;
+    }
+    return;
```

Comment:
Do we want to check for this *only* in `std.os.unlinkatW`, or should we actually move this into `std.os.windows.DeleteFile`? A Windows user is not obligated to use `std.os.unlinkatW` after all. In fact, some would probably prefer to use `std.os.windows` as the native Windows API.

At the very least, if this is not the plan, we should add a comment to that effect in `std.os.windows.DeleteFile` documenting the fact that Windows may silently fail to delete a non-empty directory, although, I think I'd personally feel more comfortable with  `DeleteFile` returning an error condition instead.

Reply:
Also, should this not be extended to `std.os.rmdirW`?

Reply:
Yea it seems you are right that the fix would be better off in `std.os.windows.DeleteFile` to always apply. Though it is a bit tilting that there are quite a few ways to do the same thing that aren't coordinated together(in regards to `std.os.rmdirW` and `std.os.unlinkW` which seem to be the same thing. `rmdir` should at least alias to `unlink` with a `remove_dir` flag.

I may address that inconsistancy as well.

Reply:
Oh, actually, `rmdir` and `unlink` are two separate calls in POSIX (and I should add that `std.os` is ultimately going to be renamed to `std.posix`). [`rmdir`](https://www.man7.org/linux/man-pages/man2/rmdir.2.html) is meant to be used to remove directories, while [`unlink`](https://www.man7.org/linux/man-pages/man2/unlink.2.html) to remove other entities like files and symlinks. It is quite unfortunate that the later added syscall `unlinkat` blurs that line. Therefore, I'd keep them separate from each other on Windows to emphasise that fact. Also, having both call `std.os.windows.DeleteFile` directly shaves off one function call if `rmdirW` was an "alias" to `unlinkW`.